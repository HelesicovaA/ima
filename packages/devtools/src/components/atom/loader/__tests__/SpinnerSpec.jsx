import React from 'react';
import { shallow } from 'enzyme';
import Loader from '../Loader';

describe('loader atom', () => {
  let wrapper = shallow(<Loader />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render title if provided', () => {
    wrapper.setProps({ title: 'Loading...' });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').text()).toBe('Loading...');
  });

  describe('defaultProps', () => {
    it('should match snapshot', () => {
      expect(Loader.defaultProps).toMatchSnapshot();
    });
  });
});
