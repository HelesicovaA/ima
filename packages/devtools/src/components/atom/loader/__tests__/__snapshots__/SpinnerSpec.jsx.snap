// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader atom defaultProps should match snapshot 1`] = `
Object {
  "title": null,
}
`;

exports[`Loader atom should match snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="loaderIcon"
  />
</div>
`;

exports[`Loader atom should render title if provided 1`] = `
<div
  className="wrapper"
>
  <div
    className="loaderIcon"
  />
  <p
    className="title"
  >
    Loading...
  </p>
</div>
`;
